### PR TITLE
Added Datastore to store comments

### PR DESCRIPTION
**Pull request by GrantComm**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.google.sps.data;
+
+/** An item on a todo list. */
```
Comment (acmcarther) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:3]: Comment looks wrong

File: portfolio/src/main/java/com/google/sps/servlets/CommentsListServlet.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String content = (String) entity.getProperty("content");
+      long timestamp = (long) entity.getProperty("timestamp");
```
Comment (acmcarther) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsListServlet.java:34]: With timestamps, I usually suggest being a bit more specific about the value unless the type itself conveys the information (i.e. when using a Timestamp type), so for this case, I'd suggest naming this "timestamp_millis".

File: portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java [hunk 3]
```diff
@@ -22,19 +22,27 @@
...
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    List<String> comments = new ArrayList<String>(); 
-    String text = getParameter(request, "commentText", "No text submitted");
-    comments.add(text);
-    
-    response.setContentType("text/html;");
-    response.getWriter().println(text);
+    String text = request.getParameter("commentText");
+    long timestamp = System.currentTimeMillis();
```
Comment (acmcarther) [inline portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java:36]: timestamp_millis here too.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -162,13 +162,46 @@ function topFunction() {
...
-async function tester() {
-  const serverReq = await fetch("/data");
-  const result = await serverReq.json();
-  console.log(result[0]);
-  console.log(result[1]);
-  console.log(result[2]);
-  document.getElementById("modalBody").innerText = result[0];
+async function loadComments() {
+  /*
+  const serverReq = await fetch("/load-comments");
```
Comment (acmcarther) [inline portfolio/src/main/webapp/script.js:167]: Delete this commented out code?

File: portfolio/src/main/webapp/index.html [hunk 5]
```diff
@@ -111,6 +111,13 @@ <h2>List of Skills</h2>
 	<button onclick="topFunction()" id="scrollToTopButton" title="Go to top">Back to Top</button>
 	<button onclick="showModal()" id="commentButton" title="Submit a Comment">Submit a Comment</button>
 
+	<section class="sectionRow" id="comments">
+		<h2 class="sectionHeader">Comments</h2>
+		<hr class="round">
+		<ul id="listofComments">
```
Comment (acmcarther) [inline portfolio/src/main/webapp/index.html:117]: minor optional nit: "commentsList" sounds marginally more idiomatic.

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -162,13 +162,46 @@ function topFunction() {
...
+}
+
+function createCommentElement(comment) {
+  const commentElement = document.createElement('li');
+  commentElement.className = 'comment';
+
+  const contentElement = document.createElement('span');
+  contentElement.innerText = comment.content;
+
+  /*
```
Comment (acmcarther) [inline portfolio/src/main/webapp/script.js:192]: Delete this too?

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 7]
```diff
@@ -1,6 +1,5 @@
 package com.google.sps.data;
 
-/** An item on a todo list. */
 public final class Comment {
```
Comment (acmcarther) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:3]: Public classes typically should have at least some javadoc. My recommendation is just a one liner.

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 8]
```diff
@@ -1,5 +1,7 @@
 package com.google.sps.data;
 
+// A comment in the comment list
+
```
Comment (acmcarther) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:4]: rm extra line

File: portfolio/src/main/java/com/google/sps/servlets/CommentsListServlet.java [hunk 9]
```diff
@@ -31,9 +31,9 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     for (Entity entity : results.asIterable()) {
       long id = entity.getKey().getId();
       String content = (String) entity.getProperty("content");
-      long timestamp = (long) entity.getProperty("timestamp");
+      long timestamp_millis = (long) entity.getProperty("timestamp");
```
[the remark this answers is not in the document]
Comment (acmcarther) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsListServlet.java:34]: This wasn't addressed?

[the remark this answers is not in the document]
Comment (GrantComm) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsListServlet.java:34]: Whoops, forgot to change it

**Pull request merged**